lib/block-supports/colors.php

```diff
@@ -76,66 +76,41 @@ function gutenberg_apply_colors_support( $block_type, $block_attributes ) {
 	$has_text_colors_support       = true === $color_support || ( is_array( $color_support ) && _wp_array_get( $color_support, array( 'text' ), true ) );
 	$has_background_colors_support = true === $color_support || ( is_array( $color_support ) && _wp_array_get( $color_support, array( 'background' ), true ) );
 	$has_gradients_support         = _wp_array_get( $color_support, array( 'gradients' ), false );
-	$classes                       = array();
-	$styles                        = array();
+	$color_block_styles            = array();
 
 	// Text colors.
 	// Check support for text colors.
 	if ( $has_text_colors_support && ! gutenberg_should_skip_block_supports_serialization( $block_type, 'color', 'text' ) ) {
-		$has_named_text_color  = array_key_exists( 'textColor', $block_attributes );
-		$has_custom_text_color = isset( $block_attributes['style']['color']['text'] );
-
-		// Apply required generic class.
-		if ( $has_custom_text_color || $has_named_text_color ) {
-			$classes[] = 'has-text-color';
-		}
-		// Apply color class or inline style.
-		if ( $has_named_text_color ) {
-			$classes[] = sprintf( 'has-%s-color', _wp_to_kebab_case( $block_attributes['textColor'] ) );
-		} elseif ( $has_custom_text_color ) {
-			$styles[] = sprintf( 'color: %s;', $block_attributes['style']['color']['text'] );
-		}
+		$preset_text_color          = array_key_exists( 'textColor', $block_attributes ) ? "var:preset|color|{$block_attributes['textColor']}" : null;
+		$custom_text_color          = isset( $block_attributes['style']['color']['text'] ) ? $block_attributes['style']['color']['text'] : null;
```

Comment:
```suggestion
		$custom_text_color          = _wp_array_get( $block_attributes, array( 'style', 'color', 'text' ), null);
```

Would this read a little easier if we used `_wp_array_get`?

If so, this would also apply to the background color and gradients as well as the typography supports.